pkg/azure/ocpgwdeployer.go

```
@@ -360,7 +349,7 @@ func (d *ocpGatewayDeployer) Cleanup(status reporter.Interface) error {
 		return status.Error(err, "deleting gateway security group failed")
 	}
 
-	err = d.deleteGateway()
+	err = d.deleteGateway(status)
 	if err != nil {
 		return status.Error(err, "removing gateway node failed")
```

`deleteGateway` now logs errors to the `status` so we should just return the error here.